esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,432 @@
+
+        IArrayOf<IEspWUResponseAttribute> & attribs = wuDetailsVisitor.getCurrentAttribs();
+        if (!attribs.empty())
+        {
+            respScope->setAttributes(attribs);
+            respScopes.append(*respScope.getClear());
+        }
+        wuDetailsVisitor.resetScope();
+    }
+    StringBuffer maxVersion;
+    maxVersion.append(wuDetailsVisitor.getMaxTimestamp());
+
+    resp.setWUID(wuid.str());
+    resp.setMaxVersion(maxVersion.str());
+    resp.setScopes(respScopes);
+}
+
+void WUDetails::buildWuScopeFilter()
+{
+    // WuScopeFilter.MaxDepth
+    wuScopeFilter.setDepth(0, requestScopeFilter.getMaxDepth());
+
+    // WuScopeFilter.Scopes
+    StringArray & scopes = requestScopeFilter.getScopes();
+    ForEachItemIn(idx1,scopes)
+        if (*scopes.item(idx1))
+            wuScopeFilter.addScope(scopes.item(idx1));
+
+    // WuScopeFilter.Ids
+    StringArray & ids = requestScopeFilter.getIds();
+    ForEachItemIn(idx2,ids)
+        if (*ids.item(idx2))
+            wuScopeFilter.addId(ids.item(idx2));
+
+    // WuScopeFilter.ScopeTypes
+    StringArray & scopeTypes = requestScopeFilter.getScopeTypes();
+    ForEachItemIn(idx3,scopeTypes)
+        if (*scopeTypes.item(idx3))
+            wuScopeFilter.addScopeType(scopeTypes.item(idx3));
+
+    // WuScopeFilter.AttributeFilter
+    buildAttribFilter(requestScopeFilter.getAttributeFilters());
+
+    // WuNestedFilter.Depth
+    wuScopeFilter.setIncludeNesting(nestedfilter.getDepth());
+    // WuNestedFilter.ScopeTypes
+    StringArray & nestedScopeTypes = nestedfilter.getScopeTypes();
+    ForEachItemIn(idx4,nestedScopeTypes)
+        if (*nestedScopeTypes.item(idx4))
+            wuScopeFilter.setIncludeScopeType(nestedScopeTypes.item(idx4));
+
+    // WUAttributeToReturn.minVersion
+    const char *minVersion = attribsToReturn.getMinVersion();
+    if (minVersion && *minVersion)
+    {
+        StringBuffer sMinVersion("version[");
+        sMinVersion.append(attribsToReturn.getMinVersion()).append("]");
+        wuScopeFilter.addFilter(sMinVersion);
+    }
+
+    // WUAttributeToReturn.Measure
+    const  char * measure = attribsToReturn.getMeasure();
+    if (measure && *measure)
+        wuScopeFilter.setMeasure(measure);
+    // WUAttributeToReturn.Attributes
+    bool isReturnAttribsSpecified = false;
+    StringArray & attribs = attribsToReturn.getAttributes();
+    ForEachItemIn(idx5,attribs)
+    {
+        if (attribs.item(idx5) && *attribs.item(idx5))
+        {
+            isReturnAttribsSpecified = true;
+            wuScopeFilter.addOutput(attribs.item(idx5));
+        }
+    }
+    IArrayOf<IConstWUScopeOverride> & overrides= attribsToReturn.getScopeOverrides();
+    ForEachItemIn(idx6, overrides)
+    {
+        IConstWUScopeOverride & cur = overrides.item(idx6);
+        const char * overrideScopeType = cur.getScopeType();
+        if (!overrideScopeType || !*overrideScopeType) continue;
```

Comment:
can the pointer be nullptr, or can it only be a null string?

Reply:
pointer check against nullptr.